docs/src/main/asciidoc/websockets-next-reference.adoc

```diff
@@ -783,6 +787,67 @@ class MyBean {
 [[websocket-next-security]]
 === Security
 
+Security capabilities are provided by the Quarkus Security extension.
+Any xref:security-identity-providers.adoc[Identity provider] can be used to convert authentication credentials on the initial HTTP request into a `SecurityIdentity` instance.
+The `SecurityIdentity` is then associated with the websocket connection.
+Authorization options are demonstrated in following sections.
+
+NOTE: When an OpenID Connect extension, `quarkus-oidc`, is used and token expires, Quarkus automatically closes connection.
+
+[[secure-http-upgrade]]
+==== Secure HTTP upgrade
+
+An HTTP upgrade is secured when a standard security annotation is placed on an endpoint class or an HTTP Security policy is defined.
+The advantage of securing HTTP upgrade is less processing, the authorization is performed early and only once.
+You should always prefer HTTP upgrade security unless, like in the example above, you need to perform an action on error or a security check based on the payload.
+
+.Use standard security annotation to secure an HTTP upgrade
+[source, java]
+----
+package io.quarkus.websockets.next.test.security;
+
+import io.quarkus.security.Authenticated;
+import jakarta.inject.Inject;
+
+import io.quarkus.security.identity.SecurityIdentity;
+import io.quarkus.websockets.next.OnOpen;
+import io.quarkus.websockets.next.OnTextMessage;
+import io.quarkus.websockets.next.WebSocket;
+
+@Authenticated <1>
+@WebSocket(path = "/end")
+public class Endpoint {
+
+    @Inject
+    SecurityIdentity currentIdentity;
+
+    @OnOpen
+    String open() {
+        return "ready";
+    }
+
+    @OnTextMessage
+    String echo(String message) {
+        return message;
+    }
+}
+----
+<1> Initial HTTP handshake ends with the 401 status for anonymous users.
+You can also redirect the handshake request on authorization failure with the `quarkus.websockets-next.server.security.auth-failure-redirect-url` configuration property.
+
+IMPORTANT: HTTP upgrade is only secured when a security annotation is declared on an endpoint class next to the `@WebSocket` annotation.
```

Comment:
This is probably a possible follow up discussion, but it appears unusual that the endpoint methods require method specific security annotations, IMHO, it would make sense to inherit the class level annotation since the security identity is bound to the connection anyway. REST endpoint users can only have a class-level security annotation and have all endpoint methods secured...

Reply:
I know it works differently (and the difference in behavior is documented), but so far I miss scenario where it would be useful. If HTTP Upgrade is permitted by class-level annotations, nothing will change because identity will not change .

Except if you need to check for example `@OnTextMessage` method arguments, that was my motivation for this PR. I believe there will be users that want to only permit access if payload is OK for given identity (and there will also be different users who will do it on a service level instead... :-) _.

I am open to discussion, but this is not a new behavior, I just moved docs to further up. Can you open a separate discussion (GH issue or discussion) if you think there is more into this, please?